src/pybind/rados/setup.py

```diff
@@ -1,51 +1,187 @@
+        return False
+    except LinkError:
+        print('\nLink Error: RADOS library not found', file=sys.stderr)
+        return False
+    else:
+        return True
+    finally:
+        shutil.rmtree(tmp_dir)
+
+
+if not check_sanity():
+    sys.exit(1)
+
+if '--without-cython' in sys.argv:
+    if not os.path.isfile('rados.c'):
+        print('Error: Cannot find Cythonized file rados.c', file=sys.stderr)
+        print('Retry without using --without-cython', file=sys.stderr)
+        sys.exit(1)
+
+
+    def cythonize(x, **kwargs):
+        return x
+
+
+    sys.argv.remove('--without-cython')
+    source = "rados.c"
+else:
+    from Cython.Build import cythonize
+
+    source = "rados.pyx"
 
 # Disable cythonification if we're not really building anything
 if (len(sys.argv) >= 2 and
-    any(i in sys.argv[1:] for i in ('--help', 'clean', 'egg_info', '--version')
-    )):
+        any(i in sys.argv[1:] for i in ('--help', 'clean', 'egg_info', '--version')
+            )):
     def cythonize(x, **kwargs):
         return x
 
+flags = get_python_flags()
+
 setup(
-    name = 'rados',
-    version = get_version(),
-    description = "Python libraries for the Ceph librados library",
-    long_description = (
+    name='rados',
+    version=__version__,
+    description="Python libraries for the Ceph librados library",
+    long_description=(
         "This package contains Python libraries for interacting with Ceph's "
-        "rados library."),
-    ext_modules = cythonize([
-        Extension("rados",
-            ["rados.pyx"],
-            libraries=["rados"],
+        "RADOS library. RADOS is a reliable, autonomic distributed object "
+        "storage cluster developed as part of the Ceph distributed storage "
+        "system. This is a shared library allowing applications to access "
+        "the distributed object store using a simple file-like interface."
+    ),
+    url='https://github.com/ceph/ceph/tree/master/src/pybind/rados',
+    license='LGPLv2+',
+    ext_modules=cythonize(
+        [
+            Extension(
+                "rados",
+                [source],
+                include_dirs=flags['cflags']['I'],
+                library_dirs=flags['ldflags']['L'],
+                libraries=["rados"] + flags['ldflags']['l'],
+                extra_compile_args=flags['cflags']['extras'] + flags['ldflags']['extras'],
             )
-    ], build_dir=os.environ.get("CYTHON_BUILD_DIR", None)),
-    cmdclass={
-        "egg_info": EggInfoCommand,
-    },
+        ], build_dir=os.environ.get("CYTHON_BUILD_DIR", None)
+    ),
+    classifiers=[
+        'Intended Audience :: Developers',
+        'Intended Audience :: System Administrators',
+        'License :: OSI Approved :: GNU Lesser General Public License v2 or later (LGPLv2+)',
+        'Operating System :: POSIX :: Linux',
+        'Programming Language :: Cython',
+        'Programming Language :: Python :: 2.6',
```

Comment:
On 05/27/2016 02:50 AM, Anirudha Bose wrote:
> In src/pybind/rados/setup.py
> <https://github.com/ceph/ceph/pull/9332#discussion_r64882338>:
>
>>              )
>> -    ], build_dir=os.environ.get("CYTHON_BUILD_DIR", None)),
>> -    cmdclass={
>> -        "egg_info": EggInfoCommand,
>> -    },
>> +        ], build_dir=os.environ.get("CYTHON_BUILD_DIR", None)
>> +    ),
>> +    classifiers=[
>> +        'Intended Audience :: Developers',
>> +        'Intended Audience :: System Administrators',
>> +        'License :: OSI Approved :: GNU Lesser General Public License v2 or later (LGPLv2+)',
>> +        'Operating System :: POSIX :: Linux',
>> +        'Programming Language :: Cython',
>> +        'Programming Language :: Python :: 2.6',
>
> I idea was to be explicit about the versions supported. I kept Python
> 2.6+ and 3.2+ since it was supported by the bindings, but it is seems
> logical to drop 2.6, 3.2, and 3.3.

Sounds good